ubloxgps/gpsparser.go

```diff
@@ -0,0 +1,137 @@
+package ubloxgps
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type GPSParser struct {
+	gpsDevice GPSDevice
+}
+
+// fix is a GPS location fix
+type Fix struct {
+	Valid     bool
+	Time      string
+	Latitude  string
+	Longitude string
+	Altitude  int32
+	Satelites int16
+}
+
+func Parser(gpsDevice GPSDevice) GPSParser {
+	return GPSParser{
+		gpsDevice: gpsDevice,
+	}
+}
+
+func (parser *GPSParser) NextFix(fix *Fix) {
```

Comment:
Can you add some docs here? What it does and how it returns values.

Also, perhaps the API would be cleaner if you just returned the struct. That's a big struct, but I think there are no performance penalties when you return it directly. It would certainly make the API a bit more straightforward: you're returning it after all.